scripts/dts/extract/globals.py

```diff
@@ -331,6 +331,58 @@ def build_cell_array(prop_array):
 
     return ret_array
 
+def child_to_parent_unmap(cell_parent, gpio_index):
+    # This function returns a (gpio-controller, pin number) tuple from
+    # cell_parent (identified as a 'nexus node', ie: has a 'gpio-map'
+    # property) and gpio_index.
+    # Note: Nexus nodes and gpio-map property are described in the
+    # upcoming (presumably v0.3) Device Tree specification, chapter
+    # 'Nexus nodes and Specifier Mapping'.
+
+    # First, retrieve gpio-map as a list
+    gpio_map = reduced[cell_parent]['props']['gpio-map']
+
+    # We now need to build the map as an array of array, which will
+    # allow to use gpio_index
+
+    # map raws are encoded as follows:
+    # [child specifier][gpio controller phandle][parent specifier]
+    # child specifier field length is connector property #gpio-cells
+    # parent specifier field length is parent property #gpio-cells
+
+    child_specifier_size = reduced[cell_parent]['props']['#gpio-cells']
+
+    # Assumption 1: We assume parent #gpio-cells is constant across
+    # the map, so we take the value of the first occurrence and apply
+    # to the whole map.
+    parent = phandles[gpio_map[child_specifier_size]]
+    parent_specifier_size = reduced[parent]['props']['#gpio-cells']
+    array_cell_size = child_specifier_size + 1 + parent_specifier_size
+
+    # Now that the length of each entry in 'gpio-map' is known,
+    # build the map as an array of arrays
+    index = 0
+    gpio_map_entries = []
+    while index < len(gpio_map):
+        gpio_map_entries.append(gpio_map[index:index+array_cell_size])
```

Comment:
Could get rid of `gpio_map_entries` at this point, and do this:

```python
for i in range(0, len(gpio_map), array_cell_size):
    entry = gpio_map[i:i+array_cell_size]
    if entry[0] == gpio_index:
        ...
        return ...

# gpio_index did not match any entry in gpio-map
return None, None
```

Reply:
Done, and reworked comments